Marlin/Configuration.h

```diff
@@ -987,6 +987,21 @@
 // Articulated robot (arm). Joints are directly mapped to axes with no kinematics.
 //#define ARTICULATED_ROBOT_ARM
 
+// For a 5 axis CNC machine in Head-Table configuration. 
+// This machine has a swivel head (B axis parallel to the Y axis) and a horizontal rotary
+// table (C axis parallel to the Z axis).
+//#define XYZBC_HEAD_TABLE
+#if ENABLED(XYZBC_HEAD_TABLE)
+  #define TOOLS 1 // Number of tools;
+  // Machine rotary zero point offset is the distance between the center of rotation of the B axis to the gauge line at the tool head.
+  #define DEFAULT_MRZP_OFFSET_MM 100.0 // (mm)
+  #define DEFAULT_TOOL_OFFSET_X { 0.0 } // (mm) Tool offset X
+  #define DEFAULT_TOOL_OFFSET_Y { 0.0 } // (mm) Tool offset Y
+  #define DEFAULT_TOOL_OFFSET_Z { 0.0 } // (mm) Gauge length (Tool length)
```

Comment:
Maybe tool related offset should be variable. I'm thinking at manual tool change...

Reply:
it already is, see variable `tool_offsets[TOOLS] which is initialized with these default values and which can be altered with G10 L1 or G10 L10. Tool length compensation is off by default and is only enabled with G43 (for 3 axis machining) or with G43.4 (RTCP for continuous 5 axis operation).

Reply:
In the long run, hotend offsets could be replaced by these, but that requires quite some refactoring to separate the concept of extruders and hotends / tools